### PR TITLE
[MRG] Improves functionality of ANT Neuro reader

### PR DESCRIPTION
**Pull request by proloyd**
Improves functionality added in [#12792](https://github.com/mne-tools/mne-python/pull/12792)  

#### What does this implement/fix?

- Adds following meta information in `RawANT.info`:
   - `'subject_info'` (`'his_id'`, `'first_name'`, `'sex'`, `'birthday'`)
   - `'device_info'` (`'type'`, `'model'`, `'serial'`, `'site'`)
   - `'meas_date'`
- Enables lazy data reading (i.e.`preload=False` argument) in `read_raw_ant()` 

#### Additional information
This requires bumping up the `antio` version to `0.3.0.dev0`.

### CODE REVIEW

**Comment by proloyd** [comment]
@mscheltienne here is the MNE pull request. Could you please review the test functions? Thanks!

File: mne/io/ant/ant.py [hunk 1]
```diff
@@ -93,6 +94,7 @@ def __init__(
         misc: str | None,
         bipolars: list[str] | tuple[str, ...] | None,
         impedance_annotation: str,
+        preload: bool | NDArray,
         *,
```
Comment (larsoner) [inline mne/io/ant/ant.py:98]:
```suggestion
        *,
        preload: bool | NDArray,
```

File: mne/io/ant/ant.py [hunk 2]
```diff
@@ -272,6 +321,7 @@ def read_raw_ant(
     misc=r"BIP\d+",
     bipolars=None,
     impedance_annotation="impedance",
+    preload=False,
     *,
```
Comment (larsoner) [inline mne/io/ant/ant.py:325]:
```suggestion
    *,
    preload=False,
```

**Review by larsoner — COMMENTED** [review]
Looks reasonable to me but I'll let @mscheltienne comment!

**Comment by proloyd** [comment]
@mscheltienne how shall we handle different datasets for testing? If I use `pytest.mark.parametrize("dataset", ["ca_208", "andy_101"])` and never mention `andy_101` explicitly, I see the following vulture error:
`mne/io/ant/tests/test_ant.py:74: unused function 'andy_101' (60% confidence)`
Does it anything to do with removal of `testing.requires_testing_data` decorator?

**Comment by mscheltienne** [comment]
I'll cut the 0.3.0 release tomorrow and will have a look at this PR then, I'm not sure I understand this last comment from my phone only.

**Comment by drammock** [comment]
> @mscheltienne how shall we handle different datasets for testing? If I use `pytest.mark.parametrize("dataset", ["ca_208", "andy_101"])` and never mention `andy_101` explicitly, I see the following vulture error: `mne/io/ant/tests/test_ant.py:74: unused function 'andy_101' (60% confidence)` Does it anything to do with removal of `testing.requires_testing_data` decorator?

I haven't looked closely as to why you need to define those extra funcs (`ca_208` and `andy_101`) and whether they specifically need to be in the test file, but FYI Vulture is finicky. It might be happier if you moved those definitions into our `conftest.py` file. If that doesn't work, you can add the func name to `tools/vulture_allowlist.py`

**Comment by mscheltienne** [comment]
I pushed a couple of changes, LGTM. If vulture complies, then the function should be added to `tools/vulture_allowlist.py` (done in the last commit). Let me know if that looks good to you!

File: mne/io/ant/ant.py [hunk 3]
```diff
@@ -132,14 +140,34 @@ def __init__(
...
+        }
+        super().__init__(
+            info,
+            preload=preload,
+            first_samps=first_samps,
+            last_samps=last_samps,
+            filenames=[fname],
+            verbose=verbose,
+            raw_extras=[raw_extras],
+        )
```
Comment (mscheltienne) [inline mne/io/ant/ant.py:170]: Is there anything we have to be careful about when `preload=False/True` and we then load `Annotations` *after* the call to `super().__init__(...)`?

Comment (proloyd) [reply mne/io/ant/ant.py:170]: To the best of my understanding from the `BaseRaw` implementation, as long as the correct `first_samps`, and `last_samps` is passed in the `super(). __init__()` call, `Annotations` reading should be good.

Comment (mscheltienne) [reply mne/io/ant/ant.py:170]: Added tests that confirm this ;) Great!

**Review by mscheltienne — APPROVED** [review]

**Comment by proloyd** [comment]
@mscheltienne  Thanks a lot!

File: mne/io/ant/ant.py (hunk 3 above)
Comment (proloyd) [reply mne/io/ant/ant.py:170]: Thanks a lot!

**Pull request merged**